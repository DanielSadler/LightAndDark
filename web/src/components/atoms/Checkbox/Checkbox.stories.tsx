import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Checkbox } from './Checkbox';

export default {
  title: 'Atoms/Checkbox',
  component: Checkbox,
} as ComponentMeta<typeof Checkbox>;

const Template: ComponentStory<typeof Checkbox> = (args) => (
  <Checkbox {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  label: 'Remember me',
  'data-testid': 'checkbox',
};
