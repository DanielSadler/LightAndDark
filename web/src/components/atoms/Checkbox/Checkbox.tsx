import React, { FC } from 'react';

type Props = React.DetailedHTMLProps<
  React.InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
> & {
  label: string;
  'data-testid'?: string;
};

export const Checkbox: FC<Props> = ({ label, ...props }) => (
  <div className="flex items-center">
    <input
      id="remember-me"
      name="remember-me"
      type="checkbox"
      className="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
      {...props}
    />
    <label
      htmlFor="remember-me"
      className="ml-2 block text-sm text-gray-900"
      data-testid={`${props['data-testid']}-label`}
    >
      {label}
    </label>
  </div>
);
