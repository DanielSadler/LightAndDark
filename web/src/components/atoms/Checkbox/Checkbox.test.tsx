import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { expect } from '@jest/globals';
import { Checkbox } from './Checkbox';

describe('Checkbox', () => {
  it('should render', () => {
    render(<Checkbox label="Test" data-testid="checkbox" />);
  });
  it('should match a snapshot', () => {
    const tree = renderer
      .create(<Checkbox label="Test" data-testid="checkbox" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
