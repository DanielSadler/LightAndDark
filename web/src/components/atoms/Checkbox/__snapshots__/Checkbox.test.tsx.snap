// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match a snapshot 1`] = `
<div
  className="flex items-center"
>
  <input
    className="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
    data-testid="checkbox"
    id="remember-me"
    name="remember-me"
    type="checkbox"
  />
  <label
    className="ml-2 block text-sm text-gray-900"
    data-testid="checkbox-label"
    htmlFor="remember-me"
  >
    Test
  </label>
</div>
`;
